src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -97,6 +97,46 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
         return this;
     }
 
+    /// <summary>
+    /// Configures the builder to match any request that meets all the criteria defined by the specified predicates.
+    /// </summary>
+    /// <param name="predicates">
+    /// Delegates to methods which must all return <see langword="true"/> for the
+    /// request to be considered a match.
+    /// </param>
+    /// <returns>
+    /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+    /// </returns>
+    /// <remarks>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </remarks>
+    public HttpRequestInterceptionBuilder For(ICollection<Predicate<HttpRequestMessage>> predicates)
```

Comment:
Sorry for the last-minute change, I think maybe these two methods would be better named `ForAll()` to make it a bit more self-descriptive and reduce the number of overloads `For()` has.